### PR TITLE
Added logic for Custom Issue title and Breadcrumbs to be sent to Sentry

### LINKED ISSUE #3: Proper Data is not sent to Sentry
Hi Jacob,

Your code is a little outdated. Lots of essential details which are needed to be sent to Sentry aren't covered in your code. I have updated the code that covers almost all those essential details. I'm not sure how I can contribute to your repo. Please suggest me how we can enhance the existing code.

Regards,
Jheel

### PR DESCRIPTION
**Pull request by jdagrawal008**
I have made enhancements in the existing code to send proper values to Sentry and also added logic with which if we want to send our custom title for any issue we can send that as well. Also, I have added logic to send breadcrumbs as well which helps in getting some proper debugs with which we can get to know the details of the exception.

### CODE REVIEW

File: force-app/main/default/classes/SYS_CustomException.cls [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ * @description       : 
+ * @author            : Jheel
+ * @group             : 
+ * @last modified on  : 04-28-2021
+ * @last modified by  : Jheel
+ * Modifications Log 
+ * Ver   Date         Author   Modification
+ * 1.0   04-28-2021   Jheel   Initial Version
+**/
+public class SYS_CustomException extends Exception {}
```
Comment (jmather) [inline force-app/main/default/classes/SYS_CustomException.cls]: Was this file accidently submitted?

File: force-app/main/default/classes/SYS_CustomException.cls-meta.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>45.0</apiVersion>
+    <status>Active</status>
+</ApexClass>
```
Comment (jmather) [inline force-app/main/default/classes/SYS_CustomException.cls-meta.xml:5]: Same here.

**Review by jmather — CHANGES_REQUESTED** [review]
Just one question around a couple files that look to have been accidentally included, but overall, amazing. Thank you!

File: force-app/main/default/classes/SYS_CustomException.cls (hunk 1 above)
Comment (jdagrawal008) [reply force-app/main/default/classes/SYS_CustomException.cls]: No. Actually, I tried to use your Sentry_Exception class itself. But it wasn't providing me with the proper message and Stacktrace. So due to that, I have to create a separate class for the Custom Exception.

File: force-app/main/default/classes/SYS_CustomException.cls-meta.xml (hunk 2 above)
Comment (jdagrawal008) [reply force-app/main/default/classes/SYS_CustomException.cls-meta.xml:5]: This is the metafile of the SYS_CustomException.cls

File: force-app/main/default/classes/SYS_CustomException.cls (hunk 1 above)
Comment (jmather) [reply force-app/main/default/classes/SYS_CustomException.cls]: I didn't see any reference to it?

Also, for the exception, did you instantiate it with the Sentry_ExceptionFactory?

So, the big deal about the Sentry_Exception and Sentry_ExceptionFactory is stack traces.

```
class CustomExceptionClass extends Exception {}
throw new CustomExceptionClass('foo');
```
is great and all, but the stack trace from that is empty.

So to counter that...

```
class CustomExceptionClass extends Sentry_Exception {}
Sentry_Exception ex = (Sentry_Exception) Sentry_ExceptionFactory.build(CustomExceptionClass.class);
ex.setMessage('foo');
```

Also - I don't see any references to it elsewhere in the code, so it looks as though the class itself is unused?

Comment (jdagrawal008) [reply force-app/main/default/classes/SYS_CustomException.cls]: Yes. It is not used anywhere in the code. It needs to be used by others to send a custom exception. Even if there is no actual system exception. 
For example, consider a use case in any project, you want to create an issue in Sentry if due to some reason a record is rejected or something. So in that scenario, we aren't running into any exception but we want to create an issue in Sentry so for this we will use this class to throw a custom exception.

So this class is not for our use but more of an outside use class. But if you feel like it's not worth having then you can remove it off. But I would suggest you, to add up some sort of notes in the readme file explaining how to use the Sentry_Exception and Sentry_ExceptionFactory classes.

**Comment by jmather** [comment]
Yeah, I need some more documentation, for sure. I'm surprised anyone has
tried to use it at all... I feel pretty avant-garde in my desire to have
Salesforce log externally... :D

**Comment by jdagrawal008** [comment]
> Yeah, I need some more documentation, for sure. I'm surprised anyone has tried to use it at all... I feel pretty avant-garde in my desire to have Salesforce log externally... :D
> […](#)

Indeed. But as I'm the one who has started with this surely people will join us too. And we can take our this package to next level soon.

**Comment by jdagrawal008** [comment]
@jmather I have added one more commit for increasing test class code coverage for the new changes which I have implemented.

**Comment by jmather** [comment]
Merged in. Thank you!

**Pull request merged**